### PR TITLE
Regenerate the Token String in the UI

### PR DESCRIPTION
**Pull request by eduardoj**
Peek 2021-09-09 15-24-Regenerate-Token

Co-authored-by: Saray Cabrera Padrón <scabrerapadron@suse.de>

### CODE REVIEW

File: src/api/app/views/webui/users/tokens/edit.html.haml [hunk 1]
```diff
@@ -8,20 +8,73 @@
...
+          = surround "'" do
+            = @token.id
+        = form_for(@token, as: :token, url: token_path, method: :put, remote: true) do |f|
+          .modal-footer
+            %a.btn.btn-sm.btn-outline-secondary.px-4{ data: { dismiss: 'modal' } }
+              Cancel
+            -# %input.btn.btn-sm.btn-danger.px-4{ type: 'submit', name: 'commit', value: 'Regenerate', data: { disable: { with: 'Regenerate' } } }
+            = f.submit('Regenerate', class: 'btn btn-sm btn-danger px-4', data: { disable: { with: 'Regenerate' } })
+
+:javascript
```
Comment (dmarcoux) [inline src/api/app/views/webui/users/tokens/edit.html.haml:64]: The JS code in this file is the same as what we have in `src/api/app/views/webui/users/tokens/new.html.haml`:

https://github.com/openSUSE/open-build-service/blob/b36fc16e3f985db371d04f5b7dbb469e846ecdcb/src/api/app/views/webui/users/tokens/new.html.haml#L54-L58

https://github.com/openSUSE/open-build-service/blob/b36fc16e3f985db371d04f5b7dbb469e846ecdcb/src/api/app/views/webui/users/tokens/new.html.haml#L79-L87

Any plans to remove this duplication?

Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/edit.html.haml:64]: > Any plans to remove this duplication?
Yes, I was on it.

> This input with the copy-to-clipboard feature could be a view component.
Ok, I will create a view component.

Comment (dmarcoux) [reply src/api/app/views/webui/users/tokens/edit.html.haml]: @eduardoj and I talked. We created the view component `CopyToClipboardInputComponent`.

File: src/api/app/views/webui/users/tokens/edit.html.haml [hunk 2]
```diff
@@ -6,22 +6,57 @@
...
     .row
       .col-12.col-md-10.col-lg-8
-        -# "as: :token" to have a consistent name across all various token classes
-        = form_for(@token, as: :token, url: token_path, method: :put, local: true) do |f|
-          .form-row#edit-token
+        = form_with(url: token_path, scope: :token, method: :put, local: true) do |f|
+          .form-row.d-none#created-token
+            .col-12.col-md-10.col-lg-9
+              %fieldset.form-group
+                = f.label(:string_readonly, 'Your new OBS Personal Access Token Secret', class: 'col-form-label col-form-label-lg')
```
Comment (hennevogel) [inline src/api/app/views/webui/users/tokens/edit.html.haml:13]: I think you should warn people that this is the only time they are going to see this...

File: src/api/app/views/webui/users/tokens/edit.html.haml [hunk 3]
```diff
@@ -6,22 +6,57 @@
...
+.modal.fade#regenerate-token-modal{ tabindex: -1, role: 'dialog', aria: { labelledby: 'regenerate-token-modal-label', hidden: true } }
+  .modal-dialog.modal-dialog-centered{ role: 'document' }
+    .modal-content
+      .modal-header
+        %h5.modal-title#regenerate-token-modal-label Do you really want to regenerate this token string?
+      .modal-body
+        %p
+          Please confirm that you want to regenerate the token string of the token with id
+          = surround "'" do
+            = @token.id
```
Comment (hennevogel) [inline src/api/app/views/webui/users/tokens/edit.html.haml:57]: The ID has no meaning to people, I would not mention it.

Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/edit.html.haml:57]: Ok, removed.

File: src/api/app/views/webui/users/tokens/edit.html.haml (hunk 2 above)
Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/edit.html.haml:13]: I added a simlar warning as it is displayed when creating a token, in the flash message.

File: src/api/app/views/webui/users/tokens/edit.html.haml (hunk 3 above)
Comment (dmarcoux) [reply src/api/app/views/webui/users/tokens/edit.html.haml:54]: The ID is still here. Did you push the changes?

Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/edit.html.haml:54]: Ops. I removed the wrong ID. Fixed.

**Comment by vpereira** [comment]
Trying to create a token without project and package displays the following error:

Screenshot_2021-09-10_10-43-56

I wonder how is it useful to the user, and what can the user do about it?

**Comment by eduardoj** [comment]
> Trying to create a token without project and package displays the following error:

True. That should be fixed now. Thanks!

**Comment by dmarcoux** [comment]
@eduardoj The PR has to be rebased on the latest version of `master`.

**Comment by eduardoj** [comment]
> @eduardoj The PR has to be rebased on the latest version of `master`.

Ops. Rebased. Thanks!

**Review by dmarcoux — APPROVED** [review]

**Pull request merged**